### PR TITLE
feat: Add EmailAddressTextField

### PR DESCRIPTION
**Pull request by FabianDevel**
The chip "shrink / expand" logic is not implemented yet

### CODE REVIEW

File: app/src/main/java/com/infomaniak/swisstransfer/ui/screen/newtransfer/ImportFilesViewModel.kt [hunk 1]
```diff
@@ -81,13 +83,19 @@ class ImportFilesViewModel @Inject constructor(
...
     //region Transfer Author Email
-    private var _transferAuthorEmail by mutableStateOf("")
-    val transferAuthorEmail = GetSetCallbacks(get = { _transferAuthorEmail }, set = { _transferAuthorEmail = it })
+    private var transferAuthorEmail by mutableStateOf("")
+    private val isAuthorEmailInvalid by derivedStateOf { !transferAuthorEmail.trim().isEmail() }
+    //endregion
+
+    //region Recipient Email
+    private var recipientEmail by mutableStateOf("")
+    private val isRecipientEmailInvalid by derivedStateOf { !recipientEmail.trim().isEmail() }
```
Comment (KevinBoulongne) [inline app/src/main/java/com/infomaniak/swisstransfer/ui/screen/newtransfer/ImportFilesViewModel.kt:92]: We could factorize these 2 `.trim().isEmail()`.

**Review by KevinBoulongne — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/infomaniak/swisstransfer/ui/screen/newtransfer/importfiles/ImportFilesScreen.kt [hunk 2]
```diff
@@ -261,57 +261,69 @@ private fun FilesToImport(
...
         Column {
-
-            val isError = transferAuthorEmail.get().isNotEmpty() && !transferAuthorEmail.get().isEmail()
-            val supportingText: @Composable (() -> Unit)? = if (isError) {
-                { Text(stringResource(R.string.invalidAddress)) }
-            } else {
-                null
-            }
+            val isAuthorError = emailTextFieldCallbacks.checkEmailError(isAuthor = true)
+            val isRecipientError = emailTextFieldCallbacks.checkEmailError(isAuthor = false)
```
Comment (KevinBoulongne) [inline app/src/main/java/com/infomaniak/swisstransfer/ui/screen/newtransfer/importfiles/ImportFilesScreen.kt:288]:
```suggestion
            val isAuthorError = checkEmailError(isAuthor = true)
            val isRecipientError = checkEmailError(isAuthor = false)
```

File: app/src/main/java/com/infomaniak/swisstransfer/ui/screen/newtransfer/importfiles/ImportFilesScreen.kt [hunk 3]
```diff
@@ -261,57 +261,69 @@ private fun FilesToImport(
...
+                supportingText = getEmailError(isRecipientError),
             )
             Spacer(Modifier.height(Margin.Medium))
         }
     }
 }
 
+@Composable
+private fun getEmailError(isAuthorError: Boolean): @Composable (() -> Unit)? {
+    val supportingText: @Composable (() -> Unit)? = if (isAuthorError) {
```
Comment (KevinBoulongne) [inline app/src/main/java/com/infomaniak/swisstransfer/ui/screen/newtransfer/importfiles/ImportFilesScreen.kt:318]:
```suggestion
private fun getEmailError(isError: Boolean): @Composable (() -> Unit)? {
    val supportingText: @Composable (() -> Unit)? = if (isError) {
```

**Review by KevinBoulongne — CHANGES_REQUESTED** [review]

**Comment by LouisCAD** [comment]
I extracted the state management of `EmailAddressTextField` into a new private class creatively named `EmailAddressTextFieldState`, to simplify the `EmailAddressTextField` function so that it's easier to grasp. It was more than 100 lines long and contained many local functions that were used only once.

I think the PR is good enough now.

**Review by LouisCAD — APPROVED** [review]

**Review by sirambd — APPROVED** [review]

**Pull request merged**